### ISSUE TITLE
Incorrect type deduction of nested/higher-level Callable with TypeVarTuple parameter

### ISSUE THREAD

**Issue by timabroad** [microsoft/pyright#6316]
**Describe the bug**
Type deduction seems to fail for function parameters that are higher-level functions (ie the parameter is a Callable that takes another Callable as a parameter) when the most nested Callable uses a TypeVarTuple. Curiously it seems to work as long as the TypeVarTuple only needs to match a single type, but if multiple types must be matched the error message indicates it's only matching a single type.

**Code or Screenshots**
Consider the following code:
```python
from typing import Callable, TypeVarTuple, Unpack, reveal_type

ArgTs = TypeVarTuple('ArgTs')

def test(fn: Callable[[Unpack[ArgTs]], None]) -> tuple[Unpack[ArgTs]]: ...
def cb(a: bytes, b: str) -> None: ...
reveal_type(test(cb)) # Correctly deduced as tuple[bytes, str]

def test2(g: Callable[[Callable[[Unpack[ArgTs]], None]], None]) -> tuple[Unpack[ArgTs]]: ...
def reg(cb: Callable[[bytes], None]) -> None: ...
def reg2(cb: Callable[[bytes, int], None]) -> None: ...
reveal_type(test2(reg))  # Correctly deduced as tuple[bytes]
reveal_type(test2(reg2)) # Incorrectly deduced as tuple[bytes]
```

Pyright output is as follows:
```
test.py
  test.py:8:13 - information: Type of "test(cb)" is "tuple[bytes, str]"
  test.py:13:13 - information: Type of "test2(reg)" is "tuple[bytes]"
  test.py:14:19 - error: Argument of type "(cb: (bytes, int) -> None) -> None" cannot be assigned to parameter "g" of type "((*ArgTs@test2) -> None) -> None" in function "test2"
    Type "(cb: (bytes, int) -> None) -> None" cannot be assigned to type "((bytes) -> None) -> None"
      Parameter 1: type "(bytes) -> None" cannot be assigned to type "(bytes, int) -> None"
        Type "(bytes) -> None" cannot be assigned to type "(bytes, int) -> None"
          Function accepts too many positional parameters; expected 1 but received 2 (reportGeneralTypeIssues)
  test.py:14:13 - information: Type of "test2(reg2)" is "tuple[bytes]"
1 error, 0 warnings, 3 informations
```

Mypy seems to behave as expected in this case:
```
test.py:8: note: Revealed type is "tuple[builtins.bytes, builtins.str]"
test.py:13: note: Revealed type is "tuple[builtins.bytes]"
test.py:14: note: Revealed type is "tuple[builtins.bytes, builtins.int]"
```

**VS Code extension or command-line**
Using `pyright 1.1.334` at the command line as well as in VS using `Pylance language server 2023.7.10 (pyright 74d8f3c4)`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.335, which I just published. It will also be included in a future release of pylance.

**Comment by timabroad** [comment]
Awesome, thanks very much Eric!
